Source/Server/Packets/ExistingPlayer.c

```diff
@@ -109,21 +103,48 @@ void receive_existing_player(server_t* server, player_t* player, stream_t* data)
     }
 
     free(lowerCaseName);
-    int       count = 0;
+
+    // ensure the player has a unique name.
+    int       found_match = 0;
     player_t *connected_player, *tmp;
-    HASH_ITER(hh, server->players, connected_player, tmp)
-    {
-        if (is_past_join_screen(connected_player) && connected_player->id != player->id) {
-            if (strcmp(player->name, connected_player->name) == 0) {
-                count++;
+    char      new_name[17] = "";
+    strcpy(new_name, player->name);
+    while (1) {
+        found_match = 0;
+        HASH_ITER(hh, server->players, connected_player, tmp)
+        {
+            if (!is_past_join_screen(connected_player) || connected_player->id == player->id) {
+                // nonapplicable
+                continue;
+            }
+
+            if (strcmp(new_name, connected_player->name) == 0) {
+                // found match
+                found_match = 1;
             }
         }
+
+        if (!found_match) {
+            // non-conflicting name. we're done here.
+            break;
+        }
+
+        new_name[0] = '\0';
+        strcpy(new_name, player->name);
+        
+        char id_str[4];
+        snprintf(id_str, 4, "%d", player->id + 15);
```

Comment:
`%d` -> `%u`, I know it's very unlikely to overflow, but if it does then the number with a minus sign would occupy 4 characters and the null terminator won't fit.